### PR TITLE
refactor(components): Add semantic colors to typography on web [97200]

### PR DESCRIPTION
**Pull request by Aiden-Brine**
## Motivations

It would be desirable for developers to be able to pass any type of semantic color to the typography component

## Changes

Various colors added to `TextColors.css` based on semantic.css and in turn `TextColors.css.d.ts` was automatically updated. I also added a snapshot test with a semantic colour.

### Added
~31~ 44 semantic colors added and a snapshot test

### Changed

N/A

### Deprecated

N/A

### Removed

N/A

### Fixed

N/A

### Security

N/A

## Testing

Open up story book and play around with different semantic text colours

Changes can be
[tested via Pre-release](https://github.com/GetJobber/atlantis/actions/workflows/trigger-qa-build.yml)

---

[In Atlantis we use Github's built in pull request reviews](https://help.github.com/en/articles/about-pull-request-reviews).

Random photo of Atlantis

### CODE REVIEW

**Comment by Aiden-Brine** [comment]
@chris-at-jobber I added all of the semantic colours but if there are any that don't make sense for text and you would like removed please let me know

File: packages/components/src/Typography/css/TextColors.css [hunk 1]
```diff
@@ -50,34 +54,218 @@
   color: var(--color-text--secondary);
 }
 
+.textReverse {
```
Comment (chris-at-jobber) [inline packages/components/src/Typography/css/TextColors.css:57]: Boggles my mind that we didn't have these colors with "text" in their name already in here 😅

**Review by chris-at-jobber — COMMENTED** [review]
@Aiden-Brine some tests have conflicts but the changes you've made look good! Can give a formal re-review once those conflicts are resolved

**Review by chris-at-jobber — COMMENTED** [review]
Design approved ✅

File: packages/components/src/Typography/css/TextColors.css [hunk 2]
```diff
@@ -50,34 +54,218 @@
...
+  color: var(--color-critical--surface);
+}
+
+.criticalOnSurface {
+  color: var(--color-critical--onSurface);
+}
+
 .warning {
   color: var(--color-warning--onSurface);
 }
```
[the remark this answers is not in the document]
Comment (chris-at-jobber) [reply packages/components/src/Typography/css/TextColors.css:93]: It definitely _is_ weird* but similar to the [mobile PR](https://github.com/GetJobber/atlantis/pull/2033), the plan for this PR is just adding more colors to the foundational Typography components. There is a [ticket for mobile](https://jobber.atlassian.net/browse/JOB-105200) that I think we could expand the scope of to include web?

***TLDR:** `warning`, `informative`, and formerly `success` and `critical`, would fail WCAG AA contrast ratios when used in isolation on top of `surface`, so at the time I thought it would be smarter to use the `onSurface` values as the default, but now that we are adding the actual onSurface values, it is much weirder
